docs/protocol/src/crypto/tct.md

```diff
@@ -0,0 +1,383 @@
+# Tiered Note Commitment Tree
+
+## Motivation and Relevant Context
```

Comment:
Hmm, the point of doing the design writeup in #491 was to already have a writeup of the problem context and motivation -- was there a reason not to include it here?